File: VB/CustomSeparator/MyMailMergeExportForm.vb
Imports System
Imports DevExpress.Snap.Core.Forms
Imports DevExpress.Snap.Forms
Imports DevExpress.XtraRichEdit.API.Native

' ...
Namespace CustomSeparator

    Public Partial Class MyMailMergeExportForm
        Inherits MailMergeExportRangeForm

        Public Sub New(ByVal controllerParameters As MailMergeExportFormControllerParameters)
            MyBase.New(controllerParameters)
            InitializeComponent()
        End Sub

        Public Enum SeparatorTypeEnum
            Html = 0
            RichText = 1
        End Enum

        Public ReadOnly Property SeparatorType As SeparatorTypeEnum
            Get
                Return CType(radioGroup1.SelectedIndex, SeparatorTypeEnum)
            End Get
        End Property

        Public ReadOnly Property HtmlSeparator As String
            Get
                Return htmlSeparatorForm.Result
            End Get
        End Property

        Public ReadOnly Property RichSeparator As Document
            Get
                Return richSeparatorForm.Result
            End Get
        End Property

        Private Sub simpleButton3_Click(ByVal sender As Object, ByVal e As EventArgs)
            radioGroup1.SelectedIndex = 0
            htmlSeparatorForm.ShowDialog()
        End Sub

        Private Sub simpleButton4_Click(ByVal sender As Object, ByVal e As EventArgs)
            radioGroup1.SelectedIndex = 1
            richSeparatorForm.ShowDialog()
        End Sub

        Private htmlSeparatorForm As HtmlSeparatorForm = New HtmlSeparatorForm()

        Private richSeparatorForm As RichSeparatorForm = New RichSeparatorForm()
    End Class
End Namespace

File: CS/CustomSeparator/MyMailMergeExportForm.cs
using System;
using DevExpress.Snap.Core.Forms;
using DevExpress.Snap.Forms;
using DevExpress.XtraRichEdit.API.Native;
// ...

namespace CustomSeparator {
    public partial class MyMailMergeExportForm : MailMergeExportForm {
        public MyMailMergeExportForm(MailMergeExportFormControllerParameters controllerParameters)
            : base(controllerParameters) {
            InitializeComponent();
        }

        public enum SeparatorTypeEnum { Html = 0, RichText = 1 };
        public SeparatorTypeEnum SeparatorType { get { return (SeparatorTypeEnum)radioGroup1.SelectedIndex; } }
        public string HtmlSeparator { get { return htmlSeparatorForm.Result; } }
        public Document RichSeparator { get { return richSeparatorForm.Result; } }

        private void simpleButton3_Click(object sender, EventArgs e) {
            this.radioGroup1.SelectedIndex = 0;
            htmlSeparatorForm.ShowDialog();
        }

        private void simpleButton4_Click(object sender, EventArgs e) {
            this.radioGroup1.SelectedIndex = 1;
            richSeparatorForm.ShowDialog();
        }

        private HtmlSeparatorForm htmlSeparatorForm = new HtmlSeparatorForm();
        private RichSeparatorForm richSeparatorForm = new RichSeparatorForm();
    }
}
